backend/clap/src/main/java/zpi/ppea/clap/service/RatingService.java

```diff
@@ -0,0 +1,51 @@
+package zpi.ppea.clap.service;
+
+import data_store.RatingType;
+import lombok.RequiredArgsConstructor;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import zpi.ppea.clap.dtos.NewRatingDto;
+import zpi.ppea.clap.dtos.PartialRatingDto;
+import zpi.ppea.clap.dtos.RatingDto;
+import zpi.ppea.clap.dtos.RatingsSubmissionResponseDto;
+import zpi.ppea.clap.mappers.DtoMapper;
+import zpi.ppea.clap.mappers.PartialRatingMapper;
+import zpi.ppea.clap.mappers.RatingMapper;
+import zpi.ppea.clap.repository.RatingsRepository;
+import zpi.ppea.clap.security.FirebaseAgent;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class RatingService {
+
+    private final RatingsRepository ratingsRepository;
+
+    @Cacheable("submissionRatings")
```

Comment:
I was thinking about cache retention policy, i think we should go with TTL 24h and invalidating cache on requests on write endpoints. Please, unless you have better idea please implement that policy with cache of your choice

Reply:
ok I will do it